lasagne/layers/input.py

```diff
@@ -67,6 +67,9 @@ def __init__(self, shape, input_var=None, name=None, **kwargs):
             if input_var.ndim != ndim:
                 raise ValueError("shape has %d dimensions, but variable has "
                                  "%d" % (ndim, input_var.ndim))
+            broad_pattern = [s == 1 for s in self.shape]
+            input_var = T.patternbroadcast(input_var, broad_pattern)
```

Comment:
> # find layer associated with input_var1
> input_layer1 = next(layer for layer in lasagne.layers.get_all_layers(network)
>                     if getattr(layer, 'input_var', None) is input_var1)

The code indeed would break after this PR, but could we have an extra parameter for the `InputLayer`, which by default is False and would broadcast the `Variable` if only set to True by the user? The above example wouldn't break. Also we can suggest in the error message from the `MergeLayer` in case if the input happens to have a `1` in shape and the MergeLayer fails because of a dimension mismatch. I confirmed with @nouiz that it is not possible to do a broadcasting without building a graph